### ISSUE TITLE
Signup with an invalid password gives username error message

### ISSUE THREAD

**Issue by vchaddha** [aws-amplify/amplify-js#5478]
**Describe the bug**
In continuation to this issue:
https://github.com/aws-amplify/amplify-js/issues/4177

I am still seeing this issue, wherein if I enter an invalid password, I am getting the error:
Username could not be created. Please make sure that the username you have entered is between 1 and 128 characters.

I am still getting the error with the following versions:
```
    "aws-amplify": "^3.0.6",
    "aws-amplify-angular": "^5.0.6",
    "aws-sdk": "^2.655.0"
```
**Error: Username could not be created. Please make sure that the username you have entered is between 1 and 128 characters.**

Whereas the error should say something about Password not being valid.

**To Reproduce**

1. Use authenticator component
2. Click on 'SignUp user'
3. Enter all other fields correctly so that they validate properly
4. Enter an invalid password - for me I just have a limitation on password length to be at least 8 characters, so I entered less than 8 chars in password.

**Expected behavior**
Displayed error message should say something about Password not being valid

<details>
  <summary><strong>Environment</strong></summary>

```
  System:
    OS: Windows 10 10.0.18362
    CPU: (8) x64 Intel(R) Core(TM) i5-8300H CPU @ 2.30GHz
    Memory: 7.23 GB / 15.80 GB
  Binaries:
    Node: 12.16.1 - C:\Program Files\nodejs\node.EXE     
    npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD        
  Browsers:
    Edge: 44.18362.449.0
    Internet Explorer: 11.0.18362.1
  npmPackages:
    angular-responsive-tables: ^0.4.4 => 0.4.4 
    angular-utils-pagination: ^0.11.1 => 0.11.1 
    aws-amplify: ^3.0.6 => 3.0.6 
    aws-amplify-angular: ^5.0.6 => 5.0.6 
    aws-sdk: ^2.655.0 => 2.655.0 
    bootstrap: ^4.4.1 => 4.4.1 
    codelyzer: ~4.5.0 => 4.5.0
    core-js: ^2.5.4 => 2.6.11
    font-awesome: ^4.7.0 => 4.7.0
    jasmine-core: ~2.99.1 => 2.99.1
    jasmine-spec-reporter: ~4.2.1 => 4.2.1
    karma: ~4.0.0 => 4.0.1
    karma-chrome-launcher: ~2.2.0 => 2.2.0
    karma-coverage-istanbul-reporter: ~2.0.1 => 2.0.6
    karma-jasmine: ~1.1.2 => 1.1.2
    karma-jasmine-html-reporter: ^0.2.2 => 0.2.2
    ng2-search-filter: ^0.5.1 => 0.5.1
    ngx-bootstrap: ^2.0.5 => 2.0.5
    bootstrap: ^4.4.1 => 4.4.1 
    codelyzer: ~4.5.0 => 4.5.0
    core-js: ^2.5.4 => 2.6.11
    font-awesome: ^4.7.0 => 4.7.0
    jasmine-core: ~2.99.1 => 2.99.1
    jasmine-spec-reporter: ~4.2.1 => 4.2.1
    karma: ~4.0.0 => 4.0.1
    karma-chrome-launcher: ~2.2.0 => 2.2.0
    karma-coverage-istanbul-reporter: ~2.0.1 => 2.0.6
    karma-jasmine: ~1.1.2 => 1.1.2
    karma-jasmine-html-reporter: ^0.2.2 => 0.2.2
    ng2-search-filter: ^0.5.1 => 0.5.1
    ngx-bootstrap: ^2.0.5 => 2.0.5
    ngx-pagination: ^5.0.0 => 5.0.0
    protractor: ~5.4.0 => 5.4.3
    rxjs: 6.0.0 => 6.0.0
    ts-node: ~7.0.0 => 7.0.1
    tslib: ^1.9.0 => 1.11.1
    tslint: ~5.11.0 => 5.11.0
    zone.js: ~0.8.26 => 0.8.29
```

</details>

**Sample code**
<amplify-authenticator [signUpConfig]="signUpConfig" [hide]="['Greetings']"></amplify-authenticator>

**Comment by amhinson** [comment]
Similar to: https://github.com/aws-amplify/amplify-js/issues/5313

Looks like a fix was pushed for this recently. Can you try using `aws-amplify@unstable`?

**Comment by amhinson** [comment]
We just released a new version (3.0.9) that should address this issue (among other things). Give it a go and let us know if anything else comes up 👍
